lib/c-api/src/wasm_c_api/module.rs

```diff
@@ -32,6 +33,30 @@ pub unsafe extern "C" fn wasm_module_new(
 #[no_mangle]
 pub unsafe extern "C" fn wasm_module_delete(_module: Option<Box<wasm_module_t>>) {}
 
+#[no_mangle]
+pub unsafe extern "C" fn wasm_module_validate(
+    store_ptr: Option<NonNull<wasm_store_t>>,
```

Comment:
I commented elsewhere on this PR about the implied semantics of `NonNull`, `store: &wasm_store_t` is probably better, especially if we're calling `unwrap` on it